### DISCUSSION TITLE
Studio components refactoring and cleanup

### DISCUSSION THREAD

**Discussion by Ziinc** [supabase/supabase#16912] [Contribute to Supabase]
This issue tracks the component library cleanup. Goal is to eliminate the `components/to-be-cleaned` folder, cleanup `packages/ui`.

Each component needs:
- prop typings (no any)
- storybook 

`packages/ui`, cleanup based on [this PR](https://github.com/supabase/supabase/pull/12427/files):
- [ ] Accordion
- [ ] Alert
- [ ] Badge
- [ ] remove Breadcrumb
- [ ] Button
- [ ] Card
- [ ] Checkbox
- [ ] Collapsible
- [ ] Context Menu
- [ ] Divider
- [ ] Dropdown
- [ ] Form
- [ ] GlassPanel
- [ ] Image
- [ ] Icon
- [ ] IconPanel
- [ ] Input
- [ ] InputNumber
- [ ] remove InputOld
- [ ] remove Introduction and related assets
- [ ] Listbox
- [ ] Loading
- [ ] Menu
- [ ] Popover
- [ ] Radio
- [ ] remove RadioOld
- [ ] Select
- [ ] SidePanel
- [ ] remove Space
- [ ] Tabs
- [ ] Textarea
- [ ] ThemeProvider
- [ ] Toast
- [ ] Toggle
- [ ] Typography
- [ ] remove Spinner
- [ ] remove lib/utilities
- [ ] cleanup lib/theme

Requires in-depth studio code refactoring:
- [ ] ChartsRenderer to split into BarChart and AreaChart under `ui/Charts`
  - [x] AreaChart
  - [ ] BarChart 
    - [x] New component + storybook
    - [ ] Removal of deprecated component
- [ ] TooltipListener hook into `<Tooltip>` component

I will add to the list as we go along.

**Comment by nnaemekaibe** [comment]
@Isaiah-Hamilton I'll like to start working on this.

**Comment by MildTomato** [comment]
@nnaemekaibe feel free to work on. 

This is actually part of quite a large refactor project since we have a lot of components in the `components/to-be-cleaned` so we might close this issue in favour of concentrating efforts on more surface level improvements.

I can keep this issue open if you do intend to do the 'to do' items above ?
Sounds like it's getting storybook examples set up for charts and improving prop typings

**Comment by nnaemekaibe** [comment]
@MildTomato Please keep it open so i can handle the todo items.

**Comment by nnaemekaibe** [comment]
Hello @MildTomato the PR's for these components are ready, before pushing i need to clarify something. I noticed that i'm supposed to work on the LineChart in ChartsRenderer, but there's no LineChart component in there, what i've got is the BarChart and AreaChart. Is it okay to assume that it's the AreaChart @Ziinc mean't and not LineChart.

**Comment by gitstart-supabase** [comment]
We're working on this issue.
